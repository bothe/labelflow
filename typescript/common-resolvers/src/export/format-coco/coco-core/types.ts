import { DbLabel } from "../../../types";

export type CocoLicense = {
  name: string;
  id: number;
  url: string;
};

export type CocoInfo = {
  contributor: string;
  date_created: string;
  description: string;
  url: string;
  version: string;
  year: string;
};

export type CocoCategory = {
  id: number;
  name: string;
  supercategory: string;
};
type Polygon = number[];

export type CocoAnnotation = {
  id: number;
  image_id: number;
  category_id: number;
  segmentation: Polygon[];
  area: number;
  bbox: [x: number, y: number, width: number, height: number];
  iscrowd: 0 | 1;
};

export type CocoImage = {
  id: number;
  width: number;
  height: number;
  file_name: string;
  license: number;
  coco_url: string;
  labelflow_url?: string;
  date_captured: string;
};

export type CocoDataset = {
  info: CocoInfo;
  licenses: CocoLicense[];
  categories: CocoCategory[];
  images: CocoImage[];
  annotations: CocoAnnotation[];
};

export type CacheLabelClassIdToCocoCategoryId = Map<string, number>;

export type DbLabelWithImageDimensions = DbLabel & {
  imageDimensions: { width: number; height: number };
};

export type DbLabelWithImageDimensionsAndLabelClass = Omit<
  DbLabelWithImageDimensions,
  "labelClassId"
> & {
  labelClassId: string;
};
