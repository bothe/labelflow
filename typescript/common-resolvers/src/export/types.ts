import {
  ExportOptionsCoco,
  ExportOptionsYolo,
  ExportOptionsCsv,
} from "@labelflow/graphql-types";
import { Context } from "../types";

export type ExportFunction<
  TOptions extends ExportOptionsCoco | ExportOptionsYolo | ExportOptionsCsv
> = (datasetId: string, options: TOptions, context: Context) => Promise<Blob>;
