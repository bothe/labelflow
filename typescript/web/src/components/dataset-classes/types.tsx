export type DatasetClassesQueryResult = {
  dataset: {
    id: string;
    name: string;
    labelClasses: {
      id: string;
      index: number;
      name: string;
      color: string;
    }[];
  };
};

export type LabelClassWithShortcut = {
  id: string;
  index: number;
  name: string;
  color: string;
  shortcut: string;
  labelsAggregates: {
    totalCount: number;
  };
};
