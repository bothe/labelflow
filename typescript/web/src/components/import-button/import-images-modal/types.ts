import { FileError, FileWithPath } from "react-dropzone";
import { SetState } from "../../../utils/types";

/**
 * Dropped file is intermediate file which let us
 * deal with rejected and accepted files at the same time
 */
export type DroppedFile = {
  file: FileWithPath;
  errors: Array<FileError>;
};

/**
 * Dropped url
 */
export type DroppedUrl = {
  url: string;
  errors: Array<Error>;
};

export type UploadStatus = "loading" | "uploaded" | "error";

export type UploadInfo = {
  status: UploadStatus;
  error?: string;
  warnings?: string[];
};

/**
 * A lookup table containing only the status of
 * the file being uploaded
 */
export type UploadInfoRecord = Record<string, UploadInfo>;

/**
 * Setter function for the upload statuses
 */
export type SetUploadInfo = SetState<UploadInfoRecord>;
