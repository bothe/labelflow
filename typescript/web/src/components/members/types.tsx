import {
  InvitationResult,
  MembershipRole,
} from "../../graphql-types/globalTypes";

export type ChangeMembershipRole = ({
  id,
  role,
}: {
  id: string;
  role: MembershipRole;
}) => void;

export type InviteMember = ({
  role,
  email,
  workspaceSlug,
}: {
  role: MembershipRole;
  email: string;
  workspaceSlug: string;
}) => Promise<InvitationResult>;

export type RemoveMembership = (id: string) => void;
