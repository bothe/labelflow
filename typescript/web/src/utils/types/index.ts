export type { SetState } from "./set-state";
