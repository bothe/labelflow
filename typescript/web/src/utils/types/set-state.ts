import { Dispatch, SetStateAction } from "react";

export type SetState<TValue> = Dispatch<SetStateAction<TValue>>;
