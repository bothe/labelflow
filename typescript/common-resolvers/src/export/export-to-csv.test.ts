import { Context } from "../types";
import { createCsv } from "./export-to-csv";
import { MOCK_REPOSITORY, DATASET_DATA } from "./export-to-csv.fixtures";

describe("CSV export", () => {
  it("Generates proper CSV", async () => {
    expect(
      await createCsv(DATASET_DATA.id, {
        repository: MOCK_REPOSITORY,
      } as Context)
    ).toMatchSnapshot();
  });
});
