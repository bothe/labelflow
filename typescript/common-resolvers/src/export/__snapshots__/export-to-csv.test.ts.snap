// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CSV export Generates proper CSV 1`] = `
"imageId,imageName,width,height,class,xmin,ymin,xmax,ymax,imageUrl
9b8fb142-8388-11ec-a8ee-f7dcb0508f86,Image 0,1920,1080,Test Class 0,100,200,200,400,https://localhost:3000/api/downloads/85456e6a-aa57-45e0-949f-5cc853f8d125/8f47e891-3b24-427a-8db0-dab362fbe269/9b8fb142-8388-11ec-a8ee-f7dcb0508f86.jpg
a552037e-8388-11ec-965b-07327355c580,Image 1,320,240,Test Class 1,100,200,200,300,https://localhost:3000/api/downloads/85456e6a-aa57-45e0-949f-5cc853f8d125/8f47e891-3b24-427a-8db0-dab362fbe269/a552037e-8388-11ec-965b-07327355c580.jpg
"
`;
